### ISSUE TITLE
Generics extended with declare module statement aren't included

### ISSUE THREAD

**Issue by tuomokar** [timocov/dts-bundle-generator#305]
**Bug report**

When using the parameter `--inline-declare-externals`, the produced `declare module` statement can miss some imports if the statement includes types that extend another type.

**Input code**

```ts
import { RowData } from '@tanstack/react-table';

declare module '@tanstack/table-core' {
    interface TableMeta<TData extends RowData> {
        stickyFilters?: boolean;
    }
}
```

**Expected output**

```ts
// RowData is imported
import { RowData } from '@tanstack/react-table';

declare module "@tanstack/table-core" {
    interface TableMeta<TData extends RowData> {
        stickyFilters?: boolean;
    }
}
```

**Actual output**

```ts
// RowData is not imported
declare module "@tanstack/table-core" {
    interface TableMeta<TData extends RowData> {
        stickyFilters?: boolean;
    }
}
```

**Additional context**

When building, these errors appear:
```
node_modules/@tanstack/table-core/build/lib/types.d.ts(17,18): error TS2428: All declarations of 'TableMeta' must have identical type parameters.
ReactTable.d.ts(7,12): error TS2428: All declarations of 'TableMeta' must have identical type parameters.
ReactTable.d.ts(7,36): error TS2304: Cannot find name 'RowData'.
```

I made a small repository to reproduce the issue. It can be found from [here](https://github.com/tuomokar/dts-bundle-generator-inline-declare-externals).

Thank you for your time in advance!

**Comment by timocov** [comment]
A note for myself (or anyone who's concerned) about the issue: currently the tool uses only root file exports to detect whether a type/value is used or not and thus whether it should be added to the bundle (via either way e.g. importing or inlining). With `--inline-declare-externals` we need to consider all these "declare externals" as well (it might be tricky tho as collecting a list of such externals is done at the same time as processing all files, so we might need to split this into 2 phases).

**Comment by timocov** [comment]
The fix has been [published in 9.4.0 version](https://github.com/timocov/dts-bundle-generator/releases/tag/v9.4.0).
